src/ui/YouTube/YouTubeThumbnail.ts

```diff
@@ -128,8 +128,10 @@ export class YouTubeThumbnail extends Component {
     Initialization.automaticallyCreateComponentsInsideResult(element, result, {
       ResultLink: this.options.embed ? { onClick: () => this.openYoutubeIframe() } : null
     });
-
-    this.modalbox = new AccessibleModal('coveo-youtube-player', element.ownerDocument.body as HTMLBodyElement, ModalBox, {
+    const modalContainer = this.searchInterface.options.attachModalsToSearchInterface
+      ? this.searchInterface.element
+      : element.ownerDocument.body;
+    this.modalbox = new AccessibleModal('coveo-youtube-player', modalContainer, ModalBox, {
```

Comment:
Would it be possible to give the modalContainer (second arg) a default value?
If so, then if the default was `element.ownerDocument.body`, then one could reuse the same method on the SearchInterface.